Sources/Exporters/OpenTelemetryProtocol/metric/OltpHTTPMetricExporter.swift

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import OpenTelemetrySdk
+import Foundation
+
+public func defaultOltpHTTPMetricsEndpoint() -> URL {
+    URL(string: "http://localhost:4318/v1/metrics")!
+}
+
+public class OtlpHttpMetricExporter: MetricExporter {
+    let endpoint: URL
+    private let httpClient: HTTPClient
+    var pendingMetrics: [Metric] = []
+    
+    public init(endpoint: URL = defaultOltpHTTPMetricsEndpoint(), useSession: URLSession? = nil) {
+        self.endpoint = endpoint
+        if let providedSession = useSession {
+            self.httpClient = HTTPClient(session: providedSession)
+        } else {
+            self.httpClient = HTTPClient()
+        }
+    }
+    
+    public func export(metrics: [Metric], shouldCancel: (() -> Bool)?) -> MetricExporterResultCode {
+        pendingMetrics.append(contentsOf: metrics)
+        return self.flush()
+    }
+
+    public func flush() -> MetricExporterResultCode {
+        var exporterResult: MetricExporterResultCode = .success
+
+        let metrics = pendingMetrics
+        pendingMetrics = []
+        let body = Opentelemetry_Proto_Collector_Metrics_V1_ExportMetricsServiceRequest
+            .with {
+                $0.resourceMetrics = MetricsAdapter.toProtoResourceMetrics(metricDataList: metrics)
+            }
+
+        var request = URLRequest(url: endpoint)
+        request.httpMethod = "POST"
+        request.httpBody = try? body.serializedData()
```

Comment:
There's a lot of duplication between the three exporters - what can we do to minimise that? There's a maintenance overhead and just means any tweak has to be done 3 times inflating effort.